packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -20,54 +20,58 @@ import '../runner/flutter_command.dart';
-  'intl': '0.17.0-nullsafety.2',
-  'fake_async': '1.2.0-nullsafety.3',
-  'js': '0.6.3-nullsafety.3',
-  'matcher': '0.12.10-nullsafety.3',
-  'meta': '1.3.0-nullsafety.6',
-  'path': '1.8.0-nullsafety.3',
-  'pedantic': '1.10.0-nullsafety.3',
-  'pool': '1.5.0-nullsafety.3',
-  'source_maps': '0.10.10-nullsafety.3',
-  'source_map_stack_trace': '2.1.0-nullsafety.4',
-  'source_span': '1.8.0-nullsafety.4',
-  'stack_trace': '1.10.0-nullsafety.6',
-  'stream_channel': '2.1.0-nullsafety.3',
-  'string_scanner': '1.1.0-nullsafety.3',
-  'term_glyph': '1.2.0-nullsafety.3',
-  'test': '1.16.0-nullsafety.16',
-  'test_api': '0.2.19-nullsafety.6',
-  'test_core': '0.3.12-nullsafety.15',
-  'typed_data': '1.3.0-nullsafety.5',
-  'vector_math': '2.1.0-nullsafety.5',
-  // Flutter team owned nnbd deps
-  'platform': '3.0.0-nullsafety.4',
-  'file': '6.0.0-nullsafety.4',
-  'process': '4.0.0-nullsafety.4',
-  'process_runner': '4.0.0-nullsafety.5',
+  'mockito': '4.1.1', // Prevent mockito from upgrading to the source gen version.
+  'vm_service_client': '0.2.6+2', // Final version before being marked deprecated.
+  // DART TEAM OWNED NNBD DEPS
+  'archive': '">=3.0.0-nullsafety.0"',
+  'async': '">=2.5.0-nullsafety.3"',
+  'boolean_selector': '">=2.1.0-nullsafety.3"',
+  'characters': '">=1.1.0-nullsafety.5"',
+  'charcode': '">=1.2.0-nullsafety.3"',
+  'clock': '">=1.1.0-nullsafety.3"',
+  'collection': '">=1.15.0-nullsafety.5"',
+  'fake_async': '">=1.2.0-nullsafety.3"',
+  'intl': '">=0.17.0-nullsafety.2"',
+  'js': '">=0.6.3-nullsafety.3"',
+  'matcher': '">=0.12.10-nullsafety.3"',
+  'meta': '">=1.3.0-nullsafety.6"',
+  'path': '">=1.8.0-nullsafety.3"',
+  'pedantic': '">=1.10.0-nullsafety.3"',
+  'petitparser': '">=4.0.0-nullsafety.1"',
+  'pool': '">=1.5.0-nullsafety.3"',
+  'source_map_stack_trace': '">=2.1.0-nullsafety.4"',
+  'source_maps': '">=0.10.10-nullsafety.3"',
+  'source_span': '">=1.8.0-nullsafety.4"',
+  'stack_trace': '">=1.10.0-nullsafety.6"',
+  'stream_channel': '">=2.1.0-nullsafety.3"',
+  'string_scanner': '">=1.1.0-nullsafety.3"',
+  'term_glyph': '">=1.2.0-nullsafety.3"',
+  'test': '">=1.16.0-nullsafety.16"',
+  'test_api': '">=0.2.19-nullsafety.6"',
+  'test_core': '">=0.3.12-nullsafety.15"',
+  'typed_data': '">=1.3.0-nullsafety.5"',
+  'vector_math': '">=2.1.0-nullsafety.5"',
+  'xml': '">=5.0.0-nullsafety.1"',
+  // FLUTTER TEAM OWNED NNBD DEPS
+  'connectivity': '">=3.0.0-nullsafety.1"',
+  'device_info': '">=2.0.0-nullsafety.1"',
+  'file': '">=6.0.0-nullsafety.4"',
+  'platform': '">=3.0.0-nullsafety.4"',
+  'process': '">=4.0.0-nullsafety.4"',
+  'process_runner': '">=4.0.0-nullsafety.5"',
+  'url_launcher': '">=6.0.0-nullsafety.1"',
+  'video_player': '">=2.0.0-nullsafety.2"',
+  // MYSTERIES
   'path_provider': '1.6.14',
-  'video_player': '2.0.0-nullsafety.2',
-  'url_launcher': '6.0.0-nullsafety.1',
-  'connectivity': '3.0.0-nullsafety.1',
-  'device_info': '2.0.0-nullsafety.1',
   'camera': '0.6.4+5',
```

Comment:
cc @bkonyi

Reply:
@bkonyi to clarify, i was cc'ing you to ask you why you had pinned this package. (i sent a comment on the original PR yesterday too, referencing this, but maybe that got buried in the spam)

Reply:
If I recall correctly, the Camera plugin was broken at the latest version which was causing builds to fail. This was pinned to the last good version so we could land critical fixes to the tool.